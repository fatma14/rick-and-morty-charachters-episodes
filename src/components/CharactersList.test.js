import React from "react";
import CharacterList from "./CharactersList";
import renderer from "react-test-renderer";

describe("CharacterList unit tests", () => {
  it("should match snapshot", () => {
    const characters = [
      { name: "joe", id: 1 },
      { name: "jill", id: 2 },
    ];
    const component = renderer.create(
      <CharacterList characters={characters} selectCharacter={() => {}} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call selectCharacter", () => {
    const selectCharacter = jest.fn();
    const characters = [
      { name: "Jack", id: 1 },
      { name: "Jane", id: 3 },
    ];
    const component = renderer.create(
      <CharacterList
        characters={characters}
        selectCharacter={selectCharacter}
      />
    );

    component.root
      .findAllByType("button")[0]
      .props.onClick({ preventDefault: () => {} });

    expect(selectCharacter).toBeCalledWith(characters[0]);
  });
});
