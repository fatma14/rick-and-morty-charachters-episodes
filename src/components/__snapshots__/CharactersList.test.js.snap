// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList unit tests should match snapshot 1`] = `
<div
  className="listElement"
>
  <button
    className="characterName"
    onClick={[Function]}
  >
    joe
  </button>
  <button
    className="characterName"
    onClick={[Function]}
  >
    jill
  </button>
</div>
`;
